src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,792 @@
+                {
+                    // return from a in b select a;
+                    case SyntaxKind.ReturnStatement:
+                        return TryConvertIfInReturnStatement(source, (ReturnStatementSyntax)parent, out documentUpdateInfo);
+                    // foreach(var x in from a in b select a)
+                    case SyntaxKind.ForEachStatement:
+                        return TryConvertIfInForEach(source, (ForEachStatementSyntax)parent, out documentUpdateInfo);
+                    // (from a in b select a).ToList(), (from a in b select a).Count(), etc.
+                    case SyntaxKind.SimpleMemberAccessExpression:
+                        return TryConvertIfInMemberAccessExpression(source, (MemberAccessExpressionSyntax)parent, out documentUpdateInfo);
+                    // var a = new [] { from a in b select a };                    
+                    case SyntaxKind.ArrayInitializerExpression:
+                    // var a = from a in b select a
+                    case SyntaxKind.EqualsValueClause:
+                    // a = from a in b select a
+                    case SyntaxKind.SimpleAssignmentExpression:
+                    // (from a in b select a)?.ToList();
+                    case SyntaxKind.ConditionalAccessExpression:
+                    // new new List<IEnumerable<int>> { from a in q select a * a };
+                    case SyntaxKind.CollectionInitializerExpression:
+                    // new new List<int>(from a in q select a * a );
+                    case SyntaxKind.Argument:
+                        return TryReplaceWithLocalFunction(source, parent, out documentUpdateInfo);
+                    default:
+                        documentUpdateInfo = default;
+                        return false;
+                }
+            }
+
+            private bool TryConvertIfInMemberAccessExpression(
+               QueryExpressionSyntax source,
+               MemberAccessExpressionSyntax memberAccessExpression,
+               out DocumentUpdateInfo documentUpdateInfo)
+            {
+                if (memberAccessExpression.Parent is InvocationExpressionSyntax invocationExpression)
+                {
+                    // This also covers generic names (i.e. with type arguments) like 'ToList<int>'. 
+                    // The ValueText is still just 'ToList'. 
+                    switch (memberAccessExpression.Name.Identifier.ValueText)
+                    {
+                        case nameof(Enumerable.ToList):
+                            return TryConvertIfInToListInvocation(source, invocationExpression, out documentUpdateInfo);
+                        case nameof(Enumerable.Count):
+                            return TryConvertIfInCountInvocation(source, invocationExpression, out documentUpdateInfo);
+                        default:
+                            return TryReplaceWithLocalFunction(source, memberAccessExpression, out documentUpdateInfo);
```

Comment:
>TryReplaceWithLocalFunction [](start = 35, length = 27)

Why we are not trying this if TryConvertIfInToListInvocation/TryConvertIfInCountInvocation fail? For example, when there are parameters.